### DISCUSSION TITLE
Will there be a ErrMultiRows in bun ?

### DISCUSSION THREAD

**Discussion by rahul-zededa** [uptrace/bun#468] [Q&A] [answered]
I was relying on `ErrMultiRows` error on calling `(*Query) Select` in **go-pg**. Is there a way i can get the number of rows returned by postgresql in a SELECT query without using a separate COUNT query in **bun** ? I tried using `(*SelectQuery) Exec` which does return `sql.Result` but it didn't scan the rows into the model.

**Comment by vmihailenco** [comment, marked as answer]
No, it is too late for such change.

>but it didn't scan the rows into the model

You can use `q.Exec(ctx, &model)` to scan results. But I don't see how that helps to replace `ErrMultiRows`.

**Reply by rahul-zededa** [reply]
> You can use `q.Exec(ctx, &model)` to scan results. But I don't see how that helps to replace `ErrMultiRows`.

What is the type of `q` your are referring to here ? because `*SelectQuery` only accepts context as an argument`func (q *SelectQuery) Exec(ctx context.Context) (res sql.Result, err error)`

**Reply by vmihailenco** [reply]
That should be fixed in https://github.com/uptrace/bun/pull/470

**Reply by rahul-zededa** [reply]
Thanks ! That helps a lot.
